daemon/api.go

```diff
@@ -1112,12 +1112,19 @@ func trySnap(c *Command, r *http.Request, user *auth.UserState, trydir string, f
 	if !filepath.IsAbs(trydir) {
 		return BadRequest("cannot try %q: need an absolute path", trydir)
 	}
-	if !osutil.IsDirectory(trydir) {
```

Comment:
Should we really remove this? Note that the new error is in a branch that we only get into when it *is* a directory, while this check would only be triggered when it was *not* a directory. Does that mean we're not properly handling the case where try is handed a non-dir path?